### ISSUE TITLE
KFX-ZIP to AZW3 or EPUB or MOBI not working

### ISSUE THREAD

**Issue by raahulkumar339** [apprenticeharper/DeDRM_tools#2381]
I had downloaded Calibre 7.22 with DeDRM 10.0.9 with KFX Input 2.18.0 on MacBook Air (macOS Sequoia - 15.1.1).

KFX (KFX-ZIP) files are not converted and showing the error, "This book is locked by DRM. To learn more about DRM and why you cannot read or convert this book in calibre, click here". SCREENSHOT ATTACHED.

I'm sure many people must be facing the same issue. Please help.

Thank You.

<img width="1021" alt="Screenshot 2024-12-07 at 10 54 17 PM" src="https://github.com/user-attachments/assets/f43234d0-06e1-41fc-8dd3-5fcdc638316a">

**Comment by ElleKayEm** [comment]
Unfortunately books from current Kindle for Mac app do not work.  This post gives some other options: https://www.mobileread.com/forums/showthread.php?t=361503

**Comment by raahulkumar339** [comment]
I see. Can you help me convert 113 ebooks I have?

I have the latest 12th Generation Paperwhite and it has become a brick since the last 2 months. Haven’t read a single book.

all I have is KFX files.

**Comment by ElleKayEm** [comment]
The options aren't great for Kindle books right now.  You need a pre-2024 E-ink Kindle for download & transfer, an older Kindle for PC app, a 2012 or earlier E-ink Kindle, or an old Kindle for Android app running in an emulator.  All those methods are probably on shaky ground, and a few books cannot be decrypted in any case.

**Comment by raahulkumar339** [comment]
Yes I agree. I was able to do until I had the 11th generation. The moment I upgraded to 12th, it was a nightmare. Had I known earlier, I would not have bought the new device until a decryption method was found.

**Comment by BeejM** [comment]
I have the same issue as above, but no Mac involved. My Kindle is an oldie but goodie, 6th Gen from 2015. Is there anyway I can use this to my advantage and remove the DRM from the new KFX-ZIP files?  Note: I'm switching over to Kobo at Christmas, but I still own a large Kindle library I want to take with me. Thanks for you time and effort on this! Cheers, Jera

Update 1 *blush* :  Just followed the link from ElleKayEm "Unfortunately books from current Kindle for Mac app do not work. This post gives some other options: https://www.mobileread.com/forums/showthread.php?t=361503 "   Thanks! Will give it a try.

Update 2:
__Method 1: Download & transfer_ via USB_  worked for me. The book was imported into Calibre in the old AZW3 format. For me, this is a workable solution. Thank you!

**Comment by ElleKayEm** [comment]
Having that Kindle, you can avoid KFX format by downloading from the website (method 1 at the link).  Or see method 2a for getting decryptable KFX from Kindle for PC.

**Comment by BeejM** [comment]
Bad news:  AZW3 is still DRM locked. Not really sure what to do. Open to suggestions.

Update: @ElleKayEm   Thanks for the serial number suggestion. I'll keep that in mind for next time. I DID get the situation resolved on my own by dropping my Kindle for PC app back to version 2.3.70840. Then I did the usual download to PC and add to Calibre (Calibre and Plugins all up-to-date). It worked for me - really this time.  :-)   

Thanks, ElleKayEm. You're a champ.

**Comment by ElleKayEm** [comment]
Enter your Kindle's serial number into DeDRM's customization.

**Comment by giulioprisco** [comment]
This worked for me:
- But ebook from Amazon
- Download ebook to a spare Kindle
- Install KFX plugins to Calibre
- Upgrade DeDRM tools to latest beta: "The latest beta is v10.0.9, as a release candidate for v10.1.0. It [can be downloaded here](https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.9)."
https://github.com/noDRM/DeDRM_tools?tab=readme-ov-file
- Add books to Library in Calibre (device view). With the latest DeDRM tools beta the book is added to library in kfx (not kfx-zip) format, and can be converted.

**Comment by LainX84** [comment]
Unfortunately, I have a 6th generation Kindle Paperwhite and the comics I bought on Amazon (which would have a color part) only downloads them to me in B&W. While the previous method (download to USB) was great and useful for eventual backup.
I also tried with an Android phone, but the file remains KFX-ZIP and therefore encrypted.

**Comment by ElleKayEm** [comment]
If they were published before Apr 23, use Kindle for PC.  See method 2a here: https://www.mobileread.com/forums/showthread.php?t=361503

**Comment by LainX84** [comment]
> If they were published before Apr 23, use Kindle for PC. See method 2a here: https://www.mobileread.com/forums/showthread.php?t=361503

Unfortunately, no, it came out later.

**Comment by mrkapqa** [comment]
but how download the book onto Kindle Device without a Wi-Fi connection?

thanks

**Comment by ElleKayEm** [comment]
The options now are older Kindles (with wifi) or Kindle for PC.  If it's a recently published book from Kindle for PC 2.8.0, you'll need DeDRM 10.0.15.
